File: administration.php
<?php

include_once("./vehicle.php");

class Administration
{
    // Define variables.
    private $Vehicles = [];

    // Add vehicle function.
    public function AddVehicle($vehicle)
    {
        if ($this->FindVehicle($vehicle->GetLicensePlate()) == true) {
            array_push($this->Vehicles, $vehicle);
            return "<br> Vehicle <span style='font-style:italic;'>\"" . $vehicle->GetName() . "\"</span> is added to the system.";
        } else {
            return "<br> Vehicle <span style='font-style:italic;'>\"" . $vehicle->GetName() . "\"</span> already exists in the system.";
        }
    }

    // Return vehicle function.
    public function ReturnVehicle($vehicle, $needsCleaning, $distance)
    {
        return $vehicle->ReturnVehicle($vehicle, $needsCleaning, $distance);
    }

    // Index vehicle function.
    public function IndexVehicle()
    {
        $i = 1;
        foreach ($this->Vehicles as $vehicle) {
            echo "<br> (" . $i . ") " . $vehicle->GetName();
            $i++;
        }
    }

    // Rent vehicle function.
    public function RentVehicle($vehicle)
    {
        return $vehicle->RentVehicle();
    }

    // Find vehicle function.
    public function FindVehicle($licensePlate)
    {
        foreach ($this->Vehicles as $vehicle) {
            if ($licensePlate == $vehicle->GetLicensePlate()) {
                return false;
            }
        }
        return true;
    }
}

File: python/administration.py
from dataclasses import dataclass
from vehicle import Vehicle

@dataclass
class Administration:

    Vehicles = []

    # Add vehicles function.
    def AddVehicle(self,vehicle):
        if self.FindVehicle(vehicle.GetLicensePlate()) == True:
            self.Vehicles.append(vehicle)
            return "Voertuig is toegevoegd."
        else:
            return "Voertuig bestaat al."

    # Rent vehicle function.
    def RentVehicle(self, vehicle):
        return vehicle.RentVehicle()

    # Return vehicle function.
    def ReturnVehicle(self, vehicle, needscleaning, distance):
        return vehicle.ReturnVehicle(vehicle, needscleaning, distance)

    # Vehicles list function.
    def IndexVehicle(self):
        i = 1
        for vehicle in self.Vehicles:
            print(f"({i}) {vehicle.GetName()}")
            i = i + 1

    # Find vehicle function.
    def FindVehicle(self, licenseplate):
        for vehicle in self.Vehicles:
            if licenseplate == vehicle.GetLicensePlate():
                return False
        return True   
